### DISCUSSION TITLE
pre|post destructive deployment without a manifest

### DISCUSSION THREAD

**Discussion by Karla-Dell** [forcedotcom/cli#2618] [Ideas]
Hello. We're trying to make use of pre-destructive-changes on the same deployment command, but we're facing errors. Apparently, that's only possible when a manifest is being used for the deployment and in our case, we're deploying from a directory. 

Example:
"$componentsDeployment = sf project deploy start --source-dir $folder --wait 1 --target-org TARGET_ORG --test-level NoTestRun --pre-destructive-changes ./preDestructiveChanges/destructiveChangesPre.xml --json | ConvertFrom-Json"

And this the the error we're getting:
"All of the following must be provided when using --pre-destructive-changes: --manifest"

How can we request the ability to deploy from a directory and use pre|post destructive changes on the same command?

**Comment by mshanemc** [comment]
> How can we request the ability to deploy from a directory and use pre|post destructive changes on the same command?

This discussion is the feature request. 😄 

The quickest path to what you want would be `project generate manifest --source-dir` to get a manifest from your folder.

I'm not sure *why* the deploy command requires a manifest when using destructive, but I do know that the original `force:source:deploy` had the same requirement for destructive changes.  

I labeled this as "will do" because I'd like for us to figure out how to make it work.

**Comment by mgalalm** [comment]
Hi, I would love to see this feature implemented. Meanwhile, we're forced to conduct separate deployments, which is time-consuming, especially with a high volume of test classes. Additionally, we have to postpone the destructive deployment until the last minute due to potential dependencies. 

I'm cautious about using project generate manifest as they may introduce unforeseen discrepancies, it feels more of reverse engineering. I prefer sticking to source deployment for full deployment and using manifest for selectively deleting components.

**Comment by mgalalm** [comment]
Hi @mshanemc!  Any update on when this feature might be released? It would have a huge impact on our pipeline. Appreciate any rough timeline you can share!
